### DISCUSSION TITLE
Welcome to mir2x Discussions!

### DISCUSSION THREAD

**Discussion by etorth** [etorth/mir2x#18] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by andersgong** [comment]
Can this build with Visual Studio?

**Reply by etorth** [reply]
It used to be, recently I switched the behaivor tree to c++20 coroutine, I saw appveyor fails when compiling coroutine code.
https://ci.appveyor.com/project/etorth/mir2x

**Comment by andersgong** [comment]
https://baijiahao.baidu.com/s?id=1683879532988362541&wfr=spider&for=pc
Maybe it's ok in the VS2019 after v16.8, I'll have a try when I finish my job.

**Reply by etorth** [reply]
thanks, please checkout the appveyor setup to see how setup the VS build, this project is based on cmake.

https://raw.githubusercontent.com/etorth/mir2x/master/appveyor.yml

**Reply by andersgong** [reply]
ok, I'll have a try, thanks for your job!

**Reply by andersgong** [reply]
Is there any test data/method so that I can test the exe?

**Reply by etorth** [reply]
did you pass the build?
If you installed git LFS, then all game resource (map/monster) gets downloaded already and installed by the cmake/make.

For the database, it will generate a default one for first time run.
You can try the default account: (id, pwd) = (test, 123456)

**Reply by andersgong** [reply]
because I don't know how to use appveyor.yml on my local pc, I'll google it.

**Reply by andersgong** [reply]
there are many errors, and I had fixed some, then I got this, maybe some packages are missed by vcpkg:

“E:\games\mir\mir2x\build\package.vcxproj”(默认目标) (1) ->
“E:\games\mir\mir2x\build\ALL_BUILD.vcxproj”(默认目标) (3) ->
“E:\games\mir\mir2x\build\client\src\client.vcxproj”(默认目标) (16) ->
(Link 目标) ->
  freetype.lib(sfnt.c.obj) : error LNK2019: 无法解析的外部符号 BrotliDecoderDecompress，函数 woff2_decompress 中引用了该符号 [E:\games\mir
\mir2x\build\client\src\client.vcxproj]
  freetype.lib(ftbzip2.c.obj) : error LNK2019: 无法解析的外部符号 BZ2_bzDecompressInit，函数 FT_Stream_OpenBzip2 中引用 了该符号 [E:\games\
mir\mir2x\build\client\src\client.vcxproj]
  freetype.lib(ftbzip2.c.obj) : error LNK2019: 无法解析的外部符号 BZ2_bzDecompress，函数 ft_bzip2_file_fill_output 中引 用了该符号 [E:\game
s\mir\mir2x\build\client\src\client.vcxproj]
  freetype.lib(ftbzip2.c.obj) : error LNK2019: 无法解析的外部符号 BZ2_bzDecompressEnd，函数 ft_bzip2_file_done 中引用了 该符号 [E:\games\mi
r\mir2x\build\client\src\client.vcxproj]
  E:\games\mir\mir2x\build\client\src\Release\client.exe : fatal error LNK1120: 4 个无法解析的外部命令 [E:\games\mir\mir2x\build\
client\src\client.vcxproj]

**Reply by etorth** [reply]
I didn't see this issue previously on appveyor, I guess the bz2 is missing?
